tests/e2e/main.go

```diff
@@ -33,6 +27,23 @@ func (t *TestSet) String() string {
 	return fmt.Sprint(*t)
 }
 
+type VersionSet []string
+
+func (c *VersionSet) Set(value string) error {
```

Comment:
It seems that this function never returns an error. Also, a trick to create sets is to use `map[string]struct{}{}`. It allows to filter out duplicates in 0(1). [Here](https://github.com/sainoe/interchain-security/blob/087a9fe65be483e08447768854c7cc71c8629b4e/tests/e2e/main.go#L190) is an example in ICS.